### PR TITLE
89 implement update area and previous button during creation

### LINKED ISSUE #89: Implement Update AREA and Previous button during creation
**Short description:**
Implement Update AREA api call and Previous button during creation.
Add a way to delete an existing area.

**Describe the solution you'd like**
Create two new API call which will delete and update an area.
Add this API call inside the source code.
Create a previous button which will be usefull to navigating between Creation step.

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose

The purpose of this pull request is to add the Update/Delete area operation.

Fixes #89 

## Added features

* You can now delete and Update areas from the API and the Flutter APP !
* A way to enable and disable AREA in Flutter app have been added.
* A previous button is now available inside the AREA creation in Flutter app.

## Updated features

* The Prisma Schema have been updated to add two new relations configuration : onDelete and onUpdate

## Fixed feature

* The "latency" apparition of AREA and Service in the Home page is now fully fixed ! (Some black magic, don't ask how)

### CODE REVIEW

File: server/server_app.js [hunk 1]
```diff
@@ -495,6 +495,82 @@ app.get(
   }
 )
 
+/**
+ * Post function used for deleting an area
+ * body.id -> id of the AREA to delete
+ * Road protected by a JWT token
```
Comment (tbellicha) [inline server/server_app.js]: Not `Road`, but `Route`

File: server/server_app.js [hunk 2]
```diff
@@ -495,6 +495,82 @@ app.get(
...
+/**
+ * Post function used for updating an area
+ * body.id -> id of the AREA to update
+ * body.name -> Name of the area
+ * body.isEnable -> Status of the area
+ * body.actionId -> Action id (optionnal if reactionId is set)
+ * body.actionParameters -> Action parameters (optional)
+ * body.reactionId -> Reaction id (optionnal if actionId is set)
+ * body.reactionParameters -> Reaction parameters (optionnal)
+ * Road protected by a JWT token
```
Comment (tbellicha) [inline server/server_app.js]: Not `Road`, but `Route`

**Review by tbellicha — CHANGES_REQUESTED** [review]

File: server/server_app.js (hunk 1 above)
Comment (LucasTesnier) [reply server/server_app.js]: :eyes:

File: server/server_app.js (hunk 2 above)
Comment (LucasTesnier) [reply server/server_app.js]: :eyes:

**Review by tbellicha — DISMISSED** [review]

File: server/server_app.js [hunk 3]
```diff
@@ -495,6 +495,82 @@ app.get(
...
+        })
+      })
+
+      await database.prisma.UsersHasActionsReactions.update({
+        where: { id: req.body.id },
+        data: {
+          name: req.body.name,
+          isEnable: req.body.isEnable,
+          Action: { connect: { id: req.body.actionId } },
+          ActionParameters: { create: ActionParameters },
```
Comment (ZQUEMA) [inline server/server_app.js]: If you create here, does the old ones are removed ?

File: server/server_app.js [hunk 4]
```diff
@@ -495,6 +495,82 @@ app.get(
...
+          Reaction: { connect: { id: req.body.reactionId } },
+          ReactionParameters: { create: ReactionParameters }
```
Comment (ZQUEMA) [inline server/server_app.js]: Same question as above

**Review by ZQUEMA — COMMENTED** [review]
Answer to my questions first.

File: server/server_app.js (hunk 3 above)
Comment (LucasTesnier) [reply server/server_app.js]: I have no idea, is it automagic ?

Comment (ZQUEMA) [reply server/server_app.js]: Didn't you test it :eyes: ?

File: server/server_app.js [hunk 5]
```diff
@@ -495,6 +495,103 @@ app.get(
...
+app.post(
+  '/api/delete/area',
+  passport.authenticate('jwt', { session: false }),
+  async (req, res, next) => {
+    if (!req.user) return res.status(401).send('Invalid token')
+    try {
+      await database.prisma.UsersHasActionsReactions.delete({
+        where: { id: req.body.id }
+      })
+      return res.status(200).send('AREA succesfully deleted.')
```
Comment (tbellicha) [inline server/server_app.js]: successfully

File: server/server_app.js [hunk 6]
```diff
@@ -495,6 +495,103 @@ app.get(
...
+      await database.prisma.UsersHasActionsReactions.update({
+        where: { id: req.body.id },
+        data: {
+          name: req.body.name,
+          isEnable: req.body.isEnable,
+          Action: { connect: { id: req.body.actionId } },
+          Reaction: { connect: { id: req.body.reactionId } }
+        }
+      })
+      return res.status(200).send('AREA succesfully updated.')
```
Comment (tbellicha) [inline server/server_app.js]: successfully

**Review by tbellicha — CHANGES_REQUESTED** [review]

File: server/server_app.js (hunk 5 above)
Comment (LucasTesnier) [reply server/server_app.js]: x)

File: server/server_app.js (hunk 6 above)
Comment (LucasTesnier) [reply server/server_app.js]: x)

**Review by tbellicha — DISMISSED** [review]

File: server/server_app.js [hunk 7]
```diff
@@ -495,6 +495,103 @@ app.get(
...
+            id: req.body.id
+          },
+          include: {
+            ActionParameters: true,
+            ReactionParameters: true
+          }
+        }
+      )
+
+      console.log(oldArea)
```
Comment (ZQUEMA) [inline server/server_app.js]: Logs here

**Review by ZQUEMA — APPROVED** [review]
One log left

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
One log left

File: server/server_app.js (hunk 7 above)
Comment (LucasTesnier) [reply server/server_app.js]: ...qdsoinfZDJOBEF

**Review by tbellicha — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**